api/proquest/importer.py

```diff
@@ -930,16 +959,18 @@ def _get_feeds(self):
         """
         self._logger.info("Started fetching ProQuest paged OPDS 2.0 feeds")
 
-        feed_temporary_files = self._download_feed_pages()
+        feed_pages_directory = tempfile.mkdtemp()
+        feed_page_files = self._download_feed_pages(feed_pages_directory)
```

Comment:
Moving the `._download_feed_pages` call into the `try` block below catch download errors and allow the `finally` block to clean up the `feed_pages_directory`.

Reply:
Totally makes sense, thank you for noticing this.